app/org/sagebionetworks/bridge/play/controllers/ConsentController.java

```diff
@@ -176,7 +170,7 @@ private Result giveConsentForVersion(int version, SubpopulationGuid subpopGuid)
         
```

Comment:
Okay I've cleaned this up so that:
- controller creates the criteriacontext (or the service for the participantService where the context is not based on the request, it's based on looking up information about the user being manipulated);
- the service updates the user's state if sharing should be turned off on a withdrawal
- the session update service will reflect this change when updating the session (again, if withdrawing)